### PR TITLE
Improved test robustness in case of failure.

### PR DESCRIPTION
**Pull request by anton7811**
Added finalizer for npu.hostif_dataplane to avoid hanging on
the test failure due to running ptf_nn and open socket.

Signed-off-by: Anton Putria <anton7811@gmail.com>

### CODE REVIEW

File: tests/ut/test_hostif_ut.py [hunk 1]
```diff
@@ -20,20 +20,26 @@ def sai_hostif_obj(npu):
     return hostif_oid
 
 
+@pytest.fixture(scope="function")
+def hostif_dataplane(request, npu):
+    hostifs = {
+        "100": "Ethernet0"
+    }
+    request.addfinalizer(npu.hostif_dataplane_stop)
```
Comment (andriy-kokhan) [inline tests/ut/test_hostif_ut.py:28]: In all other places we use `yield` for the teardown. If possible, I'd prefer to follow the same approach across the TCs.

Comment (anton7811) [reply tests/ut/test_hostif_ut.py:28]: fixed.

**Review by andriy-kokhan — APPROVED** [review]

File: tests/ut/test_hostif_ut.py [hunk 2]
```diff
@@ -20,20 +20,26 @@ def sai_hostif_obj(npu):
     return hostif_oid
 
 
+@pytest.fixture(scope="function")
+def hostif_dataplane(request, npu):
```
Comment (andriy-kokhan) [inline tests/ut/test_hostif_ut.py:24]: I think `request` can be removed now.

Comment (anton7811) [reply tests/ut/test_hostif_ut.py]: done

**Review by andriy-kokhan — APPROVED** [review]

**Pull request merged**